File: d66-social/src/main/java/com/eroldme/d66/subreddit/post/Post.java
package com.eroldme.d66.subreddit.post;

import com.eroldme.d66.appuser.AppUser;
import com.eroldme.d66.subreddit.subreddit.Subreddit;
import lombok.*;
import org.springframework.lang.Nullable;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;

import java.time.Instant;

import static javax.persistence.FetchType.LAZY;

/**
 * @author Mompati 'Patco' Keetile
 * @created 19-12-2022 @ 11:54
 */
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
public class Post {
  @Id
  @SequenceGenerator(name = "post_id_seq", sequenceName = "post_id_seq", allocationSize = 1)
  @GeneratedValue(generator = "post_id_seq", strategy = GenerationType.IDENTITY)
  private Long postId;
  @NotBlank(message = "Post Name cannot be empty or Null.")
  private String post;
  @Nullable
  private String url;
  @Nullable
  @Lob
  private String description;
  private Integer voteCount;
  @ManyToOne(fetch = LAZY)
  @JoinColumn(name = "userId", referencedColumnName  = "userId")
  private AppUser appUser;
  private Instant createdDate;
  @ManyToOne(fetch = LAZY) @JoinColumn(name ="id", referencedColumnName = "id")
  private Subreddit subreddit;
}

File: d66-social/src/main/kotlin/com/eroldme/d66/subredit/post/Post.kt
package com.eroldme.d66.subredit.post

import com.eroldme.d66.subredit.subredit.Subreddit
import com.eroldme.d66.user.User
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.NoArgsConstructor
import org.springframework.lang.Nullable
import java.time.Instant
import javax.persistence.*
import javax.persistence.FetchType.LAZY
import javax.validation.constraints.NotBlank

/**
 * @author Mompati 'Patco' Keetile
 * @since 06-12-2022 @ 13:57
 *
 */
@Data
@Entity
@Builder
@AllArgsConstructor
@NoArgsConstructor
class Post(
    @Id
    @SequenceGenerator(name = "post_id", sequenceName = "post_id", allocationSize = 1)
    @GeneratedValue(generator = "post_id", strategy = GenerationType.IDENTITY)
    private val postId: Long,
    @NotBlank(message = "Post Name cannot be empty or Null.")
    private val postName: String,
    @Nullable
    private val url: String,
    @Nullable
    @Lob
    private val description: String,
    private val voteCount: Int,
    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "userId", referencedColumnName  = "userId")
    private val user: User,
    private val createdDate: Instant,
    @ManyToOne(fetch = LAZY) @JoinColumn(name ="id", referencedColumnName = "id")
    private val subreddit: Subreddit
)